src/System.ServiceModel.Tests.Common/src/testproperties.targets

```diff
@@ -0,0 +1,167 @@
+                    case '\a': literal.Append(@"\a"); break;
+                    case '\b': literal.Append(@"\b"); break;
+                    case '\f': literal.Append(@"\f"); break;
+                    case '\n': literal.Append(@"\n"); break;
+                    case '\r': literal.Append(@"\r"); break;
+                    case '\t': literal.Append(@"\t"); break;
+                    case '\v': literal.Append(@"\v"); break;
+                    default:
+                        if (c >= 0x20 && c <= 0x7e)
+                        {
+                            // ASCII printable character
+                            literal.Append(c);
+                        }
+                        else
+                        {
+                            // As UTF16 escaped character
+                            literal.Append(@"\u");
+                            literal.Append(((int)c).ToString("x4"));
+                        }
+                        break;
+                }
+            }
+            
+            string setStatement = String.Format("            properties[\"{0}\"] = \"{1}\";", propertyName, literal.ToString());
+            sb.AppendLine(setStatement);
+        }
+            
+	      sb.AppendLine("        }");
+	      sb.AppendLine("    }");
+	      sb.AppendLine("}");
+        
+        // Do not rewrite file if there are no changes.  This permits incremental builds.
+        string contentsToWrite = sb.ToString();
+        string priorContents = File.Exists(FileName) ? File.ReadAllText(FileName) : String.Empty;
+        if (!String.Equals(contentsToWrite, priorContents))
+        {
+            Log.LogMessage("Test properties were generated to {0}.", FileName);
+            File.WriteAllText(FileName, sb.ToString());
+        }
+        else
+        {
+             Log.LogMessage("No changes were made to the existing file {0}", FileName);
+        }
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
+  <!--
+    CreateTestProperties:
+      This target generates a file to initialize the TestProperties dictionary
+      with a set of known name/value pairs.  The file will also contain
+      generated static strings for each property name.  Code accessing the
+      properties in TestProperties.Properties should use these generated
+      names rather than separate string literals.
+  -->
+  <Target Name="CreateTestProperties">
+    <!-- 
+      This item collection gives the name and value for each property to generate.
+      New items can be added here or in any other project or targets file.
+      The Value metadata contains the value to use, and DefaultValue contains
+      the value to use if Value is empty.  This permits properties to be
+      specified externally to override the defaults here.
+      Example: 'msbuild /p:UseFiddlerUrl=true' will override the default false.
+    -->
+    <ItemGroup>
+      <TestProperty Include="BridgeResourceFolder">
```

Comment:
Every property here is just a suggestion.  We could add, change or remove any of these.  I expect the list to grow.